libs/server/Resp/RespCommand.cs

```diff
@@ -460,47 +460,7 @@ static RespCommand MatchedNone(RespServerSession session, int oldReadHead)
-                                                    {
-                                                        readHead += 8;
-                                                        count -= 1;
-                                                        return (RespCommand.BITOP, (byte)BitmapOperation.OR);
-                                                    }
-                                                }
-                                                // 3-character operations
-                                                else if (remainingBytes > length + 6 + 9)
-                                                {
-                                                    if (*(uint*)(ptr + 11) == MemoryMarshal.Read<uint>("$3\r\n"u8))
-                                                    {
-                                                        // Optimistically adjust read head and count
-                                                        readHead += 9;
-                                                        count -= 1;
-
-                                                        if (*(ulong*)(ptr + 12) == MemoryMarshal.Read<ulong>("3\r\nAND\r\n"u8) || *(ulong*)(ptr + 12) == MemoryMarshal.Read<ulong>("3\r\nand\r\n"u8))
-                                                        {
-                                                            return (RespCommand.BITOP, (byte)BitmapOperation.AND);
-                                                        }
-                                                        else if (*(ulong*)(ptr + 12) == MemoryMarshal.Read<ulong>("3\r\nXOR\r\n"u8) || *(ulong*)(ptr + 12) == MemoryMarshal.Read<ulong>("3\r\nxor\r\n"u8))
-                                                        {
-                                                            return (RespCommand.BITOP, (byte)BitmapOperation.XOR);
-                                                        }
-                                                        else if (*(ulong*)(ptr + 12) == MemoryMarshal.Read<ulong>("3\r\nNOT\r\n"u8) || *(ulong*)(ptr + 12) == MemoryMarshal.Read<ulong>("3\r\nnot\r\n"u8))
-                                                        {
-                                                            return (RespCommand.BITOP, (byte)BitmapOperation.NOT);
-                                                        }
-
-                                                        // Reset read head and count if we didn't match operator.
-                                                        readHead -= 9;
-                                                        count += 1;
-                                                    }
-                                                }
-                                            }
+                                            return (RespCommand.BITOP, 0);
```

Comment:
Why was this change reverted? Our goal going forward is to try and consolidate all the command and subcommand parsing into the parsing phase, rather than depend on individual operations to parse. That way ACL can also be managed centrally after the (command, subcommand) are known.

Reply:
Whereas the most data commands I have seen follow the format <command> <key>, BITOP expects the key after the operation type (i.e. AND,OR, XOR etc.) which could be considered as subcommand, though technically it isn't.
I made the change in order to deal with the event the provided operation type is not valid.
I reverted it back to the original state and added a workaround.
Now if BITOP is matched and then the subcommand is not matched we still end up in BITOP but capture the error using BitmapOperation.NONE flag and drain the buffer.